Update DNSSEC & SecondaryDNS RFC

onlyhavecans opened the pull request:
A customer wrote in about the technical accuracy of this article. While reviewing it, I noticed the RFC was finalized in 2020. 
The first purpose of this PR is to fix that.

Offline record signing was mentioned as another solution to address this, but we do not support that at this time, and I am not fully able to vouch for whether signed non-existence is fully solved for pre-signed AXFR. We could either note this or look into this as a possibility but before that...

While reviewing this article, should we rephrase or replace it as it applies explicitly to DNSimple? This reasonably technical document explains potential issues around something we don't support. I wonder if explaining "the why" is as valuable to our customers. It could become more about clarifying what we support and what customers can do depending on their use case.
Before I refactored, I wanted to bring up the conversation.


I'm CCing @m0rcq on this as a second pair of high-technical eyes.

stephanoodle commented:
NGL, the Groove conversation was too technical for me to understand to verify anything with it. 

What I will say is, that linking the correct RFC in the documentation is a good move. The documentation should also be clear on what we do and do not support at DNSimple.

itsalyse commented:
Do we have numbers for how many customers are actually clicking through to/finding this article?

onlyhavecans commented:
> Do we have numbers for how many customers are actually clicking through to/finding this article? 

I do not have those numbers; maybe @sbastn can offer better information on the metrics.

I went ahead and updated this PR to reflect your suggestion. It seems to flow better and makes a clearer experience with fewer clicks for those who are not highly technical.

While I was there, I fixed some glaring typos in my eyeline.

sbastn commented:
> Do we have numbers for how many customers are actually clicking through to/finding this article?

Year to date, here are the numbers for this page compared to other DNSSEC and Secondary DNS pages:

- [The article](https://support.dnsimple.com/articles/dnssec-and-secondary-dns/) we're reviewing in this PR received 179 visits.
- The [DNSSEC](https://support.dnsimple.com/articles/dnssec/) article received 864 visits.
- The [Add a secondary DNS server to DNSimple](https://support.dnsimple.com/articles/secondary-dns/) article received 559 visits.

Traffic sources for this document include:
- Google: 101 referrals
- Direct: 39 referrals
- The [DNSSEC article ](https://support.dnsimple.com/articles/dnssec/): 8 referrals
- [DNSimple-app](https://dnsimple.com): 4 referrals
- The The [Add a secondary DNS server to DNSimple](https://support.dnsimple.com/articles/secondary-dns/): 4 referrals

>  I'm wondering if it might be easier to remove this doc entirely and just bake it into the warning that's on the secondary and DNSSEC docs

Sounds good to me. Right now we have the following alerts at the top of these docs:

<img width="702" alt="Screenshot 2024-08-02 at 08 41 57" src="https://github.com/user-attachments/assets/751277f8-9fac-4aed-8bd5-b460ab5d213c">
